File: result/translate/chatgpt/java2cpp/c++/SUM_AREA_RECTANGLES_POSSIBLE_ARRAY.cpp
int f_gold(int a[], int n) {
    int *copy = new int[n];
    std::copy(a, a + n, copy);
    std::sort(copy, copy + n);
    int sum = 0;
    bool flag = false;
    int len = 0;
    for (int i = 0; i < n - 1; i++) {
        if ((copy[i] == copy[i + 1] || copy[i] - copy[i + 1] == 1) && !flag) {
            flag = true;
            len = copy[i + 1];
            i++;
        } else if ((copy[i] == copy[i + 1] || copy[i] - copy[i + 1] == 1) && (flag)) {
            sum = sum + copy[i + 1] * len;
            flag = false;
            i++;
        }
    }
    delete[] copy;
    return sum;
}

File: result/translate/chatgpt/python2java/java/SUM_AREA_RECTANGLES_POSSIBLE_ARRAY.java
public static int f_gold(int[] a, int n) {
    Arrays.sort(a, 0, n);
    int sum = 0;
    boolean flag = false;
    int len = 0;
    int i = 0;
    while (i < n - 1) {
        if ((a[i] == a[i + 1] || a[i] - a[i + 1] == 1) && !flag) {
            flag = true;
            len = a[i + 1];
            i++;
        } else if ((a[i] == a[i + 1] || a[i] - a[i + 1] == 1) && flag) {
            sum += a[i + 1] * len;
            flag = false;
            i++;
        }
        i++;
    }
    return sum;
}